### PR TITLE
Kno 400

### PR DESCRIPTION
**Pull request by mikiodehartj1**
# Description

Thank you for contributing to a JupiterOne integration!

Please include a summary of the change and which issue is fixed. Please also
include relevant motivation and context. List any dependencies that are required
for this change.

## Type of change

Please leave any irrelevant options unchecked.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### General Development Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### Integration Development Checklist:

Please leave any irrelevant options unchecked.

- [ ] I have checked for additional permissions required to call any new API
      endpoints, and have documented any additional permissions in
      `jupiterone.md`, where necessary.
- [ ] My changes properly paginate the target service provider's API
- [ ] My changes properly handle rate limiting of the target service provider's
      API
- [ ] My new integration step is instrumented to execute in the correct order
      using `dependsOn`
- [ ] I have referred to the
      [JupiterOne data model](https://github.com/JupiterOne/data-model/tree/main/src/schemas)
      to ensure that any new entities/relationships, and relevant properties,
      match the recommended model for this class of data
- [ ] I have updated the `CHANGELOG.md` file to describe my changes
- [ ] When changes include modifications to existing graph data ingestion, I've
      reviewed all existing managed questions referencing the entities,
      relationships, and their property names, to ensure those questions still
      function with my changes.

### CODE REVIEW

File: src/steps/profile/converter.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+export function createCustomProfileEntity(
+  customProfile: CustomProfiles,
+): Entity {
+  return createIntegrationEntity({
+    entityData: {
+      source: customProfile,
+      assign: {
+        _key: customProfileKey(customProfile),
```
Comment (i5o) [inline src/steps/profile/converter.ts:20]:
```suggestion
        _key: `kandji_profile:${customProfile.id}`,
```

File: src/steps/profile/converter.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import { Entities } from '../constants';
+import {
+  parseTimePropertyValue,
+  createIntegrationEntity,
+  Entity,
+} from '@jupiterone/integration-sdk-core';
+
+export function customProfileKey(customProfile: CustomProfiles): string {
+  return `kandji_profile:${customProfile.id}`;
+}
```
Comment (i5o) [inline src/steps/profile/converter.ts:11]:
```suggestion
```

File: src/config.ts [hunk 3]
```diff
@@ -40,5 +40,5 @@ export async function validateInvocation(
   }
 
   const apiClient = createAPIClient(config);
-  await apiClient.verifyAuthentication();
+  //await apiClient.verifyAuthentication();
```
Comment (i5o) [inline src/config.ts:43]:
```suggestion
  await apiClient.verifyAuthentication();
```

File: src/client.ts [hunk 4]
```diff
@@ -88,6 +94,22 @@ export class APIClient {
...
+    });
+
+    const body = await this.request<CustomProfiles[]>(endpoint);
+    console.log(body);
+    const results = (body as any).results;
+    for (const result of results) {
+      await iteratee(result);
+    }
+  }
+
```
Comment (i5o) [inline src/client.ts:112]:
```suggestion
  public async iterateCustomProfiles(
    iteratee: ResourceIteratee<CustomProfiles>,
  ) {
    const endpoint = this.withBaseUri('/api/v1/library/custom-profiles', {
      page: 1
    });

    const body = await this.request<CustomProfiles[]>(endpoint);
    const results = (body as any).results;

    for (const result of results) {
      await iteratee(result);
    }
  }
```

File: src/steps/profile/index.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    await jobState.addEntity(customProfileEntity);
+    //        if (accountEntity && deviceEntity) {
+    //          await jobState.addRelationship(
+    //            createDirectRelationship({
+    //              _class: RelationshipClass.HAS,
+    //              from: accountEntity,
+    //              to: deviceEntity,
+    //            }),
+    //          );
+    //        }
```
Comment (i5o) [inline src/steps/profile/index.ts:28]:
```suggestion
```

**Comment by i5o** [comment]
@mikiodehartj1 run `npm run format` on your terminal

File: src/steps/profile/converter.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import { CustomProfiles } from '../../types';
+import { Entities } from '../constants';
+import {
+  parseTimePropertyValue,
+  createIntegrationEntity,
+  Entity,
+} from '@jupiterone/integration-sdk-core';
+
+
```
Comment (i5o) [inline src/steps/profile/converter.ts:9]:
```suggestion
```

File: src/client.ts [hunk 7]
```diff
@@ -88,6 +94,20 @@ export class APIClient {
     }
   }
 
+  public async iterateCustomProfiles(
+    iteratee: ResourceIteratee<CustomProfiles>,
+  ) {
+    const endpoint = this.withBaseUri('/api/v1/library/custom-profiles', {
+      page: 1
```
Comment (i5o) [inline src/client.ts:101]:
```suggestion
      page: 1,
```

File: src/client.ts [hunk 8]
```diff
@@ -88,6 +94,20 @@ export class APIClient {
     }
   }
 
+  public async iterateCustomProfiles(
+    iteratee: ResourceIteratee<CustomProfiles>,
+  ) {
+    const endpoint = this.withBaseUri('/api/v1/library/custom-profiles', {
+      page: 1,
```
Comment (i5o) [inline src/client.ts:101]:
```suggestion
      page: "1",
```

File: src/client.ts [hunk 9]
```diff
@@ -88,6 +94,20 @@ export class APIClient {
     }
   }
 
+  public async iterateCustomProfiles(
+    iteratee: ResourceIteratee<CustomProfiles>,
+  ) {
+    const endpoint = this.withBaseUri('/api/v1/library/custom-profiles', {
+      page: "1",
```
Comment (i5o) [inline src/client.ts:101]:
```suggestion
      page: '1',
```

**Review by dobregons — APPROVED** [review]

**Pull request merged**